android/src/main/java/com/polidea/blemulator/SimulatedAdapter.java

```diff
@@ -217,6 +217,12 @@ public void onEvent(ConnectionState newState) {
 
         dartValueHandler.addConnectionStatePublisher(deviceIdentifier, onEventCallback);
 
+        if (!knownPeripherals.containsKey(deviceIdentifier)) {
+            knownPeripherals.put(deviceIdentifier,
+                    new DeviceContainer(deviceIdentifier, "", null, null)
```

Comment:
Shouldn't there be also a name?

Reply:
User is not passing it from the library and we have no way of knowing the name without waiting for the response from Dart, which we cannot do, as a connection state callback for this peripheral will be fired first.

I assume that since it's the user themselves who's initiating the call, the cached peripheral does not need the correct name.